tiledb/common/thread_pool.h

```diff
@@ -48,6 +48,49 @@
 namespace tiledb {
 namespace common {
 
+/**
+ * A class-scope singleton kept in existence by members, allocated on first
+ * use.
+ */
+template <class T>
+struct class_singleton {
+  typedef T singleton_type;
+  typedef std::shared_ptr<T> member_type;
+  typedef std::weak_ptr<T> central_type;
```

Comment:
we don't typedef trivial types

Reply:
The singleton class is destined for use outside this compilation unit. These are library-style typedef that encapsulate the particular pointer classes used. They'll be changing to tdb-version classes at some point as well; no client code should need to be changed then.